### DISCUSSION TITLE
Enhancement Proposal - Add -ForegroundColor & -BackgroundColor operators to Write-Verbose

### DISCUSSION THREAD

**Discussion by supersalt97** [PowerShell/PowerShell#21325] [Enhancements]
Hi everyone,
I am a long-time PowerShell user and this is my first Enhancement Request/Proposal (please be gentle).

In all my scripts, I use Write-Verbose for logging/debug. This is great, as I can switch it on/off with the $VerbosePreference variable.

I used to use Write-Host, which I like, as it has the -ForgroundColor and -BackgroundColor switches (so you can take advantage of different colours for different purposes, Red for Errors, Yellow to Information, Green for OK responses etc).

Write-Verbose is missing the -ForgroundColor  and -BackgroundColor operators and I am proposing we add it as an optional switch users can use. Default can remain yellow. If users, like me want to take advantage of different colours for Verbose logging, we can.

Any thoughts, concerns or suggestions?
Obviously, I understand one of the primary reasons for Verbose Logging it to log in Transcripts/Log Files, which lack colour support, however as I run scripts hundreds of times a day, I and other users can spot "problem" Verbose output if it's in Red in the Console.

**Comment by rhubarb-geek-nz** [comment]
Rather than referring to specific colours I suggest using `$PSStyle,` for example `$PSStyle.Formatting.Verbose` default is yellow and `$PSStyle.Formatting.Error` is red.  Not every GUI environment uses a black background ( eg MacOS, CDE etc )

Consideration should also be given to abide by the [NO_COLOR](https://no-color.org/) environment variable.

**Comment by jdhitsolutions** [comment]
You can create any string with ANSI sequences.

![image](https://github.com/PowerShell/PowerShell/assets/12089920/f9d864b9-ef42-4b71-88b2-2e1af6aaf6b5)

You could also use PSStyle with the strings.
